lib/rules/no-extra-parens.js

```diff
@@ -73,6 +75,38 @@ module.exports = {
          * @private
          */
         function ruleApplies(node) {
+            if (node.type === "JSXElement") {
+                const isSelfClosing = node.openingElement.selfClosing;
+
+                // Self-closing components do not have `closingElement`, so we must check the start
+                // and end lines of this particular component directly
+                const isSingleLine = isSelfClosing
+                    ? node.loc.start.line === node.loc.end.line
+                    : astUtils.isTokenOnSameLine(node.openingElement, node.closingElement);
```

Comment:
Is there a reason to use `astUtils.isTokenOnSameLine` here rather than `node.loc.start.line === node.loc.end.line` in both cases? In other words, could you use `node.loc.start.line === node.loc.end.line` even when the element is not self-closing?